### PR TITLE
egl: Don't error check eglQueryDmaBufModifiersEXT

### PR DESCRIPTION
**Pull request by Drakulix**
Fixes NVIDIA's 520 driver.

### CODE REVIEW

File: src/backend/egl/display.rs [hunk 1]
```diff
@@ -799,7 +799,17 @@ fn get_dmabuf_formats(
...
+        //
+        // - NVIDIA proprietary since 520
+        //
+        // So lets ignore any errors of this call on purpose(!),
+        // which will let `num` stay at `0` and handle the format
+        // as unsupported by explicit modifiers.
+        // Which is probably what the error is suppose to indicate
+        // although the spec doesn't seem to demand it...
+        unsafe {
             ffi::egl::QueryDmaBufModifiersEXT(
```
Comment (cmeissl) [inline src/backend/egl/display.rs:813]: Is the only error that can be returned `EGL_BAD_PARAMETER` or are we actually interessted in other errors?
Should we explicitely handle that one error and maybe also explicitely set `num = 0` in that case or is num guaranteed to be untouched if an error occured?
Or maybe we want to at least log that error?

**Review by cmeissl — APPROVED** [review]
LGTM apart from some minor questions

File: src/backend/egl/display.rs (hunk 1 above)
Comment (Drakulix) [reply src/backend/egl/display.rs:813]: You are right, some safety percussions cannot hurt.

Does this look better?

Comment (cmeissl) [reply src/backend/egl/display.rs:813]: 👍

**Review by cmeissl — APPROVED** [review]

File: src/backend/egl/display.rs (hunk 1 above)
Comment (i509VCB) [reply src/backend/egl/display.rs:813]: The EGL specification says that the function will return EGL_FALSE on failure

File: src/backend/egl/display.rs [hunk 2]
```diff
@@ -799,7 +799,17 @@ fn get_dmabuf_formats(
...
+        // Some drivers return EGL_BAD_PARAMETER here for formats
+        // they themselves returned for the query above.. *sigh*
+        //
+        // - NVIDIA proprietary since 520
+        //
+        // So lets ignore any errors of this call on purpose(!),
+        // which will let `num` stay at `0` and handle the format
+        // as unsupported by explicit modifiers.
+        // Which is probably what the error is suppose to indicate
+        // although the spec doesn't seem to demand it...
```
Comment (i509VCB) [inline src/backend/egl/display.rs:811]: This is indeed a driver bug as it disagrees with the spec.

I checked the EGL CTS and there is only testing for whether eglGetProcAddress exposed the symbols and not whether the implementation behaves properly: https://github.com/KhronosGroup/VK-GL-CTS/search?p=2&q=EGL_EXT_image_dma_buf_import_modifiers&type=

> * If <format> is not one of the formats advertised by QueryDmaBufFormatsEXT for the same <dpy>, the error EGL_BAD_PARAMETER is generated.

Have you filed a driver bug with Nvidia @Drakulix?

I guess if we wanted to be extra sure, one of us could write a pull request for the EGL CTS and add a new test for this?

**Pull request merged**